crates/outbound-http/src/lib.rs

```diff
@@ -31,8 +33,19 @@ impl OutboundHttp {
     /// If `None` is passed, the guest module is not allowed to send the request.
     fn is_allowed(&self, url: &str) -> Result<bool, HttpError> {
         let url = Url::parse(url).map_err(|_| HttpError::InvalidUrl)?;
+
+        if self.is_same_origin(&url) {
```

Comment:
There is a possible sneaky gap here, if two applications are hosted on the same host but at different base URLs. I don't think that's possible with any current Spin implementation, so not a blocker (mentioning it mostly because this was the justification for have `base` in the first place).